### ISSUE TITLE
pow(baseNumber, 0) gives invalid results

### ISSUE THREAD

**Issue by wanderer** [indutny/bn.js#42]
Here is my test;
```javacsript
const BN = require('bn.js');
const base = new BN(256).toRed( BN.red('k256'));
const exponent = new BN(0);

const result = base.redPow(exponent)
console.log(result.toString());
```
the result is 256, when it should be 1

**Comment by dcousens** [comment]
@wanderer why not submit a PR to add this test?

**Comment by wanderer** [comment]
@dcousens sorry busy week, just added the test. its now part of  #37

**Comment by wanderer** [comment]
ok i also fixed the issue in #37

**Comment by dcousens** [comment]
thanks @wanderer  :)
